MLS.Agent/CommandLine/CommandLineParser.cs

```diff
@@ -56,6 +56,9 @@ public static class CommandLineParser
             StartServer startServer = null,
             InvocationContext context = null);
 
+        public delegate Task<int> KernelServer(
```

Comment:
Maybe `StartKernelServer` or something? We're more typically using verb names for delegate types.